docs/adr/0005-replace-apache-with-nginx-gunicorn.md

```diff
@@ -0,0 +1,130 @@
+# ADR 5: Replace Apache with Nginx + Gunicorn for Manager Service
+
+- **Author(s)**: [Mikolaj Kasprzak](https://github.com/MikolajKasprzak)
+- **Date**: 2025-10-21
+- **Status**: `Accepted`
+
+## Context
+
+The SceneScape Manager service was originally hosted using Apache with mod_wsgi, which introduced several operational and security challenges:
+
+### Key Problems:
+
+1. **Security vulnerabilities**: Apache required root privileges for initialization, creating unnecessary attack surface
+2. **Complex configuration**: Multi-layered Apache configuration with mod_wsgi, SSL termination, and proxy rules was difficult to maintain
+3. **Volume permission issues**: Dynamic UID/GID changes led to unreliable file system permissions
+4. **Deployment complexity**: Monolithic container with Apache + Django made scaling and debugging difficult
+5. **Resource overhead**: Apache's process model was heavier than needed for a Python web application
+
+### Technical Debt:
+
+- Complex initialization scripts (`scenescape-init`, `webserver-init`) with root privilege requirements
+- Dynamic user/group ID management causing permission denied errors
+- Mixed responsibilities in single container (web server + application server)
+- Legacy configuration files and unused dependencies
+
+The system needed a more secure, maintainable, and cloud-native architecture aligned with modern containerization best practices.
+
+## Decision
+
+We will replace the Apache + mod_wsgi architecture with a **Nginx + Gunicorn** separation of concerns approach:
+
+### New Architecture:
+
+#### Docker Compose Deployment:
```

Comment:
Please use mermaid diagrams to depict all deployments mentioned in the doc
- the original deployment (in the context)
- proposed compose deployment
- proposed k8s deployments
- (optionally) alternatives